src/util/kaldi-table-inl.h

```diff
@@ -164,9 +187,11 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
   }
   const T &Value() {
     StateType orig_state = state_;
-    if (state_ == kHaveScpLine) LoadCurrent();  // Takes
-    // state_ to kLoadSucceeded or kLoadFailed.
-    if (state_ == kLoadFailed) {  // this can happen due to
+    if (state_ == kHaveScpLine || state_ == kObjectLoaded) {
```

Comment:
You could rename LoadCurrent() to EnsureObjectLoaded(), and call it here regardless of 'state_'.  Then you could make it the responsibility of EnsureObjectLoaded() to return (with return-status true) if the object was already loaded and nothing more was needed to do.